src/Compilers/CSharp/Portable/Symbols/ObsoleteAttributeHelpers.cs

```diff
@@ -123,6 +147,30 @@ internal static ObsoleteDiagnosticKind GetObsoleteDiagnosticKind(Symbol symbol,
                     // later stage
                     return ObsoleteDiagnosticKind.LazyPotentiallySuppressed;
             }
+
+            static bool isExperimentalSymbol(NamedTypeSymbol namedTypeSymbol)
+            {
+                return namedTypeSymbol.Arity == 0
+                    && namedTypeSymbol.HasNameQualifier("System.Diagnostics.CodeAnalysis")
+                    && namedTypeSymbol.Name.Equals("ExperimentalAttribute", StringComparison.Ordinal);
```

Comment:
>&& namedTypeSymbol.Name.Equals("ExperimentalAttribute", StringComparison.Ordinal) [](http://example.com/codeflow?start=20&length=81)

Consider checking `Name` before `HasNameQualifier" since checking `Name` may be cheaper and rule out mismatches more often.